src/domain/cohort.js

```diff
@@ -4,19 +4,44 @@ import dbClient from '../utils/dbClient.js'
  * Create a new Cohort in the database
  * @returns {Cohort}
  */
-export async function createCohort(startDate, endDate) {
+export async function createCohort(name, users, startDate, endDate) {
   const createdCohort = await dbClient.cohort.create({
     data: {
       startDate,
-      endDate
+      endDate,
+      name,
+      users: {
+        connect: users.map((user) => {
+          return { id: user.id }
+        })
+      }
     },
     include: {
-      users: true,
+      users: {
+        select: {
+          id: true,
+          cohortId: true,
+          role: true,
+          email: true,
+          profile: {
+            select: {
+              firstName: true,
+              lastName: true,
+              bio: true,
+              githubUsername: true
+            }
+          }
+        }
+      },
       deliveryLogs: true
     }
   })
 
-  return new Cohort(createdCohort)
+  if (createdCohort) {
+    return { ...createdCohort, users: flatten(createdCohort) }
```

Comment:
a bit confused about this: can you explain the `users: flatten(createdCohort)` what does this exactly do to the createdCohort?

Reply:
Of course! The problem is that the user always returns a user that has the profile nested. However, the front end prefers to not have it nested so we need to take the nested profile out. I do this with the flatten function and then return the created cohort, with the users (with the nested profile) replaced with the non nested users. Hope that make sense! Let me know if you have further questions!

Reply:
OK - got it, flatten is a custom function that does that for you.